v-next/hardhat/test/internal/builtin-plugins/solidity/build-system/cache.ts

```diff
@@ -69,42 +70,44 @@ describe("ObjectCache", () => {
       assert.deepEqual(filesAfter, filesBefore);
     });
 
-    it("should remove everything with the max age set to 0", async () => {
+    it("should remove everything with the max age set to -1", async () => {
       const filesBefore = await getAllFilesMatching(cachePath);
       assert.notDeepEqual(filesBefore, []);
-      // NOTE: We're waiting a little so that the file's atime is different
-      await new Promise((resolve) => setTimeout(resolve, 10));
-      await cache.clean(0);
+      // NOTE: -1 is not enough on Windows for some reason. Maybe there is some
+      // delay in how the access time is set on files on creation?
+      await cache.clean(-100);
       const filesAfter = await getAllFilesMatching(cachePath);
       assert.deepEqual(filesAfter, []);
     });
 
-    it("should remove everything with the max size set to 0", async () => {
+    it("should remove everything with the max size set to -1", async () => {
       const filesBefore = await getAllFilesMatching(cachePath);
       assert.notDeepEqual(filesBefore, []);
-      await cache.clean(undefined, 0);
+      await cache.clean(undefined, -1);
```

Comment:
This is unrelated to flakiness. I just thought it would be a more accurate test as technically we could have an 0 byte entry in the cache so setting max size to 0 wouldn't remove it.